wallet/src/de/schildbach/wallet/payments/SendCoinsTaskRunner.kt

```diff
@@ -36,30 +35,43 @@ import org.dash.wallet.common.transactions.ByAddressCoinSelector
 import org.slf4j.LoggerFactory
 import javax.inject.Inject
 
+
 class SendCoinsTaskRunner @Inject constructor(
     private val walletApplication: WalletApplication
 ) : SendPaymentService {
     private val log = LoggerFactory.getLogger(SendCoinsTaskRunner::class.java)
 
-    override suspend fun sendCoins(address: Address, amount: Coin, constrainInputsTo: Address?): Transaction {
+    override suspend fun sendCoins(
+        address: Address,
+        amount: Coin,
+        constrainInputsTo: Address?,
+        emptyWallet: Boolean
+    ): Transaction {
         val wallet = walletApplication.wallet ?: throw RuntimeException("this method can't be used before creating the wallet")
         Context.propagate(wallet.context)
-        val sendRequest = createSendRequest(address, amount, constrainInputsTo)
+        val sendRequest = createSendRequest(address, amount, constrainInputsTo, emptyWallet)
```

Comment:
Since we need to first top-up our CrowdNode account address and then send the required amount from it to CrowdNode, we either need to top-up amount + fee or use `emptyWallet` property.

Reply:
Determining the fee is more involved.  

One way to include the fee is this:
1. Take the intended deposit amount and add 192 duffs (the fee for sending 1 input to one output)
2. Send that to our CrowdNode address
3. Then do `emptyWallet` to send to CrowdNode (the fee is 192 duffs/satoshis).

Not sure if the user would care about

Reply:
> The user may not care about their depositing being short my 192 duffs.

This is what I thought but then I guess it's still better to account for it. I'll change that in the withdraw story.

Reply:
That will be fine with me.